oaknational/Oak-Web-Application
# Added remaining all-years logic in curriculum downloads

## Description
Added all-years logic into threads and unit-detail pages in curriculum downloads (docx)

## Issue(s)

Fixes `CUR-981`

## How to test
Test for regressions in cycle 1. For Cycle 2

 - Enable cycle 2
 - Go to physical education
 - Test swimming (all-years) appears in threads and unit-detail pages


## Screenshots
<img width="602" alt="Screenshot 2024-10-10 at 18 18 56" src="https://github.com/user-attachments/assets/855d3e1b-6c3f-4ee9-916e-e41e32e125a1">
<img width="602" alt="Screenshot 2024-10-10 at 18 18 43" src="https://github.com/user-attachments/assets/21f92ca1-10d0-4b18-8d01-3ba5b38fc96c">
